### PR TITLE
Awesome books: refactor to use JavaScript classes

### PR DESCRIPTION
**Pull request by katfogy**
**Project Objectives**
- Use JavaScript classes.
- Understand how to use medium-fidelity wireframes to create a UI.
- I used JavaScript classes instead of objects and arrays.
- I used objects and arrays, but do it inside classes.
- Create class methods to add and remove books.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -5,55 +5,66 @@ const msg = document.getElementById('message');
...
+    this.displayBooks();
+  }
+
+  deleteBook(id) {
+    let books = JSON.parse(localStorage.getItem('storebooks')) || [];
+    books = books.filter((e) => e.id !== id);
     localStorage.setItem('storebooks', JSON.stringify(books));
+    msg.innerHTML = 'Book Deleted';
+    this.displayBooks();
   }
```
Comment (wayungi) [inline main.js:53]: -  [ ] I have noticed that once I add a few books and delete one of them when I add another book, the deleted book reappears. Kindly fix this. **To reproduce the bugs, kindly follow steps in the diagram**
books

File: main.js [hunk 2]
```diff
@@ -3,57 +3,68 @@ const bookTitle = document.getElementById('title');
...
+        title: this.bookTitle.value,
+        author: this.bookAuthor.value,
+      });
+      msg.innerHTML = 'Book Added Successfully';
+      localStorage.setItem('storebooks', JSON.stringify(books));
+    }
+    this.bookTitle.value = '';
+    this.bookAuthor.value = '';
+    this.displayBooks();
+  }
```
Comment (julie-ify) [inline main.js:45]: - [ ] There is a bug in your app
  - Add some books
  - Remove the first book
  - Add another book and remove it
  - This removes two books.

  If you inspect the localStorage you will see that the book that was added last has the same `id` as the a previously added book so when it is removed, two books of the same id are removed. Kindly fix this by ensuring each book at every point as its own unique ID.

  image

**Review by julie-ify — CHANGES_REQUESTED** [review]
Hi @katfogy and @Paul-tes,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @katfogy, and @Paul-tes👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Highly organized code. ✔️
- No linter errors. :heavy_check_mark:
- GitHub flow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**